internal/pkg/cosign/sign.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cosign
+
+import (
+	"bytes"
+	"context"
+	"crypto"
+	"encoding/base64"
+	"io"
+
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/static"
+	"github.com/sigstore/sigstore/pkg/signature"
+	signatureoptions "github.com/sigstore/sigstore/pkg/signature/options"
+)
+
+type Signer interface {
```

Comment:
This is where you really want good comments documenting the expectations.

Reply:
> I like the interface a lot more. Left a few comments inline, but at a high level:
> 
> 1. Can we shuffle the package structure around so that implementations generally live in their own packages with private types and public ctors in terms of the interface?

yed

> 2. As I was building up `oci` I was trying to ensure fairly high coverage as I went, and it should now be fairly high.  Can we try to do the same here?

Will do, but I'd like to have the package structure stable before I go implementing all the accoutrements for testing